src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncStateActor.scala

```diff
@@ -0,0 +1,72 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.SupervisorStrategy.Escalate
+import akka.actor.{Actor, ActorLogging, OneForOneStrategy, SupervisorStrategy}
+import akka.pattern.pipe
+import io.iohk.ethereum.blockchain.sync.FastSyncController.SyncState
+import io.iohk.ethereum.blockchain.sync.FastSyncStateActor.GetStorage
+import io.iohk.ethereum.db.storage.FastSyncStateStorage
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Persists current state of fast sync to a storage. Can save only one state at a time.
+  * If during persisting new state is received then it will be saved immediately after current state
+  * was persisted.
+  * If during persisting more than one new state is received then only the last state will be kept in queue.
+  */
+class FastSyncStateActor extends Actor with ActorLogging {
+
+  override val supervisorStrategy: SupervisorStrategy = OneForOneStrategy(){ case _ => Escalate }
+
+  def receive: Receive = {
+    // after initialization send a valid Storage reference
+    case storage: FastSyncStateStorage => context become idle(storage)
+  }
+
+  def idle(storage: FastSyncStateStorage): Receive = {
+    // begin saving of the state to the storage and become busy
+    case state: SyncState => persistState(storage, state)
+    //
+    case GetStorage => sender() ! storage
+  }
+
+  def busy(storage: FastSyncStateStorage, state: Option[SyncState]): Receive = {
+    // update state waiting to be persisted later. we only keep newest state
+    case state: SyncState => context become busy(storage, Some(state))
+    // exception was thrown during persisting of a state. push
+    case Failure(e) => throw e
```

Comment:
Parent actor wouldn't know what to do with this message as well. Benefit of throwing this exception is that it will be handled by a supervisor somewhere in the actor hierarchy.